src/main/java/com/google/devtools/build/lib/analysis/platform/ConstraintCollection.java

```diff
@@ -43,10 +43,17 @@
   @Nullable private final ConstraintCollection parent;
   private final ImmutableMap<ConstraintSettingInfo, ConstraintValueInfo> constraints;
 
+  /**
+   * Create a new {@link ConstraintCollection} with the given constraint values.
+   */
   public ConstraintCollection(ImmutableList<ConstraintValueInfo> constraints) {
     this(null, constraints);
   }
 
+  /**
+   * Create a new {@link ConstraintCollection} which inherits constraint values from {@code parent}.
+   * Any constraints passed in {@code constraints} will override values from {@code parent}.
```

Comment:
Nit: Either move this line to its own paragraph (prefixed with a blank line and `<p>`) or convert it to an `@params` tag.

Reply:
Done.